### PR TITLE
[Azure Storage] Support cancellation of operations during Creation phase

### PR DESCRIPTION
**Pull request by RamjotSingh**
Fixes one small part of the issue brought up in https://github.com/Azure/durabletask/issues/565

Adds support for cancellation of queuing of orchestration for Azure Storage based orchestrations.

### CODE REVIEW

File: src/DurableTask.Core/IOrchestrationServiceClient.cs [hunk 1]
```diff
@@ -28,18 +28,20 @@ public interface IOrchestrationServiceClient
...
         /// <summary>
         /// Creates a new orchestration and specifies a subset of states which should be de duplicated on in the client side
         /// </summary>
         /// <param name="creationMessage">Orchestration creation message</param>
         /// <param name="dedupeStatuses">States of previous orchestration executions to be considered while de-duping new orchestrations on the client</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
         /// <exception cref="OrchestrationAlreadyExistsException">Will throw an OrchestrationAlreadyExistsException exception If any orchestration with the same instance Id exists in the instance store and it has a status specified in dedupeStatuses.</exception>
         /// <returns></returns>
-        Task CreateTaskOrchestrationAsync(TaskMessage creationMessage, OrchestrationStatus[] dedupeStatuses);
+        Task CreateTaskOrchestrationAsync(TaskMessage creationMessage, OrchestrationStatus[] dedupeStatuses, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (cgillum) [inline src/DurableTask.Core/IOrchestrationServiceClient.cs:44]: Unfortunately, this is a breaking change in terms of binary compatibility. I think semver requires this to be done as part of introducing a new major version. I'm not sure we're ready to do that right now. @ConnorMcMahon, thoughts?

Comment (RamjotSingh) [reply src/DurableTask.Core/IOrchestrationServiceClient.cs:44]: Ah I feared as much :(.

Comment (ConnorMcMahon) [reply src/DurableTask.Core/IOrchestrationServiceClient.cs:44]: I'm open to revisiting this down the line, but I agree this is a breaking change that we aren't ready to take on (at least in the next few months).

@RamjotSingh I will reply to your issue filed to see if we can help you address your issues without making a breaking change.

Comment (RamjotSingh) [reply src/DurableTask.Core/IOrchestrationServiceClient.cs:44]: If we are just concerned with IL binding creating issues, I suppose we can add the new method as a new one (one requiring cancellation token to be passed always). I don't remember if IL is ok with that or not. One place I see it creating issues would be if people have created their own implementations of ServiceClient.

Comment (ConnorMcMahon) [reply src/DurableTask.Core/IOrchestrationServiceClient.cs:44]: In general, I don't know if cancellation tokens even make sense in the context of long-lived orchestrations that can run across multiple VMs.

In general, we support orchestration "termination" for this use case.

Comment (jviau) [reply src/DurableTask.Core/IOrchestrationServiceClient.cs:44]: @ConnorMcMahon - I agree that it doesn't make that much sense in terms of cancelling the orchestration. But maybe this makes sense for cancelling the _enqueuing_ of the orchestration, if it is not already started?

**Pull request closed without merging**